cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -660,6 +661,34 @@ public void getApplicationDetails(FullHttpRequest request, HttpResponder respond
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(result));
   }
 
+  /**
+   * Returns the source control metadata and sync status of all applications
+   * filter query format - "name=&lt;name-filter&gt; AND syncStatus=&lt;SYNCED/UNSYNCED&gt;".
+   */
+  @GET
+  @Path("/appscmstatus")
```

Comment:
I think just 'sourcecontrol' here would be better, as it would be similar to the APIs for getting all programs in a namespace:
https://github.com/cdapio/cdap/blob/bbd6f8000c9bbd4e50a821c79026dc8b10d0f6e9/cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java#L1937

Reply:
Or if the intent is to eventually support listing source control information for more than just apps, maybe a path like '/sourcecontrol/apps'?

Reply:
Noted. Done